Forum_v1/Controllers/TopicController.cs

```diff
@@ -96,27 +171,165 @@ public async Task<IActionResult> CreateNewMessage(string text, int topicId)
                 return RedirectToAction("Index", "Topic");
             }
 
-            if (text == null)
+            if (ModelState.IsValid)
+            {
+
+                Message mes = new Message()
+                {
+                    TopicId = model.TopicId,
+                    ApplicationUserId = user.Id,
+                    Date=DateTime.Now, DateOfLastEdit=DateTime.Now, isDelited=false, isEdited=false,
+                    Text = model.Text,
+                    UserName = user.Email
+                };
+
+                await _messageRepo.CreateAsync(mes);
+
+                Topic topic = await _topicRepo.FindByIdAsync(model.TopicId);
+                topic.DateOfLastMessage = mes.Date;
+                await _topicRepo.UpdateAsync(topic);
+
+                return RedirectToAction("EnterIntoTopic", "Topic", new { topic_Id = model.TopicId });
+            }
+            else 
+            {
+                return View(model);            
+            }  
+        }
+
+
+
+        [HttpGet]
+        [Authorize]
+        public async Task<ActionResult> EditMessage(int? id, int curPage)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }            
+
+            Message message = await _messageRepo.FindByIdWithIncludeUserAsync((int)id);
+
+            if (message == null)
+            {
+                return NotFound();
+            }
+
+            ApplicationUser user = await _userManager.FindByEmailAsync(User.Identity.Name);
+
+            bool isAdmin = false;
+
+            if (user != null)
             {
-                return RedirectToAction("EnterIntoTopic", "Topic", new { Id = topicId});
+                IList<string> _rolelist = await _userManager.GetRolesAsync(user);
+
+                if (_rolelist.Contains("admin"))
+                {
+                    isAdmin = true;
+                }
             }
 
-            Message mes = new Message()
-            {  
-                TopicId = topicId,
-                ApplicationUserId = user.Id,
-                Text=text,
-                UserName=user.Email
+            if (user != null && message.ApplicationUserId != user.Id && isAdmin == false)
```

Comment:
user was checked for not null above so this part can be placed inside previuos if